lib/internal/process/next_tick.js

```diff
@@ -127,7 +105,11 @@ function nextTick(callback) {
 
   if (queue.isEmpty())
     setHasTickScheduled(true);
-  queue.push(new TickObject(callback, args, getDefaultTriggerAsyncId()));
+  const asyncId = newAsyncId();
+  const triggerAsyncId = getDefaultTriggerAsyncId();
+  if (initHooksExist())
+    emitInit(asyncId, 'TickObject', triggerAsyncId, { callback, args });
+  queue.push({ callback, args, asyncId, triggerAsyncId });
```

Comment:
This is straight up misleading because the object pushed on the queue is not the same as what’s emitted from the Async Hook.

Reply:
Why does it matter? The object being pushed to the queue is not available to userland, so it's not as if an end user could be storing a reference to it and doing some kind of comparison in a hook somewhere. Additionally, the hooks are receiving the exact same pertinent information (callback and arguments -- not even a copy of the arguments). The only difference here is that the "private" details (otherwise set via internal symbols) are excluded completely, which users should not have access to to begin with.

Reply:
FYI, I think this is fine. We want to move away from the `resource` object exposing internal API anyway. It is just that it is technically difficult because the `_owner` property isn't set at handle initialization.

Reply:
I'm also not sure this really matters for nextTicks, which are non-interactable (i.e. uncancelable).

Reply:
@mscdex can you change this block to:

```js
if (initHooksExist()) {
   const tock = { callback, args, asyncId, triggerAsyncId };
   queue.push(tock);
   emitInit(asyncId, 'TickObject', triggerAsyncId, tock);
 } else {
   queue.push({ callback, args, asyncId, triggerAsyncId });
 }
```

I checked and I'm getting:

```
 process/next-tick-breadth-args.js n=4000000        ***     28.56 %       ±1.06% ±1.42% ±1.86%
 process/next-tick-breadth.js n=4000000             ***     73.76 %       ±1.76% ±2.35% ±3.07%
 process/next-tick-depth-args.js n=12000000                 -0.47 %       ±1.20% ±1.60% ±2.08%
 process/next-tick-depth.js n=12000000                      -1.43 %       ±1.61% ±2.14% ±2.78%
 process/next-tick-exec-args.js n=5000000           ***      3.09 %       ±1.53% ±2.03% ±2.65%
 process/next-tick-exec.js n=5000000                ***      5.67 %       ±2.18% ±2.91% ±3.80%
```

This should also resolve @apapirovski objection.

Reply:
@mscdex Does @mcollina's suggestion seem good to you? 

@apapirovski Would that change resolve your concern?

Reply:
Yup, if those results hold then my objection goes away.

Reply:
I stand by what I said previously. Also I do not think making the async ID properties more public (moving from hidden behind an internal symbol to being available via a normal property) is a good idea.